### ISSUE TITLE
Question about error: Cannot invoke "String.length()" because "this.input" is null

### ISSUE THREAD

**Issue by mattmazzola** [Azure/autorest.java#1282]
When I try to generate a Java client using a simple swagger file autorest fails with this message: 

### Exception 
```
warning | PreCheck/CheckDuplicateSchemas | Checking for duplicate schemas, this could take a (long) while.  Run with --verbose for more detail.
verbose | [5.04 s]   ModelerFour/api-version-mode:client
verbose | [5.04 s]   ModelerFour/api-version-parameter:constant
verbose | [5.04 s]   ModelerFour/use-model-namespace:false
fatal   | Postprocessor | Unhandled error: Cannot invoke "String.length()" because "this.input" is null
java.lang.NullPointerException: Cannot invoke "String.length()" because "this.input" is null
        at java.base/java.net.URI$Parser.parse(URI.java:3165)
        at java.base/java.net.URI.<init>(URI.java:623)
        at com.azure.autorest.postprocessor.Postprocessor.processInternal(Postprocessor.java:55)
        at com.azure.autorest.extension.base.plugin.NewPlugin.process(NewPlugin.java:202)
        at com.azure.autorest.postprocessor.Main.lambda$main$1(Main.java:12)
        at com.azure.autorest.extension.base.jsonrpc.Connection.lambda$dispatch$2(Connection.java:151)
        at com.azure.autorest.extension.base.jsonrpc.Connection.lambda$process$3(Connection.java:270)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)

fatal   | Process() cancelled due to failure
error   |   Error: Plugin postprocessor reported failure.
```

### Swagger input

I originally thought it was a problem with my swagger.json input and some model validation, but even for a simple file with no schemas defined it still fails

```
{
  "openapi": "3.0.1",
  "info": {
    "title": "Test Swagger",
    "version": "v1"
  },
  "paths": {
    "/api/v1/health": {
      "get": {
        "tags": [
          "Health"
        ],
        "operationId": "GetHealth",
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    }
  },
  "components": {
    "schemas": {}
  }
}
```

### Command executed:
```bash
autorest --java --input-file:swaggerEdit.json --output-folder:java-client --namespace:TestNamespace --verbose
```

Perhaps this is an issue with latest version, and I may work around by specifying specific version?

I saw documentation allow specifying version by `--use:@autorest/java@4.0.x`.

**Comment by weidongxu-microsoft** [comment]
I don't think current autorest supports OpenAPI 3.0

**Comment by mattmazzola** [comment]
I would hope if version 3.x was unsupported the tool would simply read the version and say so explicitly instead of giving java exception from processing malformed data and us having to infer it is the version which is the issue.

Nevertheless, I tried serializing as v2 with
```
app.UseSwagger(options =>
{
    options.SerializeAsV2 = true;
});
```

https://docs.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-swashbuckle?view=aspnetcore-6.0&tabs=visual-studio#add-and-configure-swagger-middleware

And get the new swagger output:
```json
{
  "swagger": "2.0",
  "info": {
    "title": "Test Swagger",
    "version": "v1"
  },
  "paths": {
    "/api/v1/health": {
      "get": {
        "tags": [
          "Health"
        ],
        "operationId": "GetHealth",
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    }
  },
  "definitions": {}
}
```

but I get the same error

![image](https://user-images.githubusercontent.com/2856501/145445531-882fc2e9-1efe-451c-bdc0-4a523923b9aa.png)

I feel like this should be a simple process given the size of audience but I'm not sure what I could change. I will look for alternatives

**Comment by mattmazzola** [comment]
I think we are going with 

https://github.com/OpenAPITools/openapi-generator

which seems to be where this is all transitioning to.  It also supports java and latest swager specs.

I think the error is probably some environmental configuration. Will close.

**Comment by weidongxu-microsoft** [comment]
@mattmazzola 
Sure. If you are using Jersey or OkHttp as client, openapi-generator would be the best util.

Tested with local build (`mvn package -Plocal`) as well, no problem on my side.

```
autorest --input-file=swagger.json --output=. --namespace=org.test --java --use=.
AutoRest code generation utility [cli version: 3.4.2; node: v14.17.5]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ~3.1.0.

There is a new version of AutoRest available (3.5.1).
 > You can install the newer version with with npm install -g autorest@latest

   Loading AutoRest core      'C:\Users\weidxu.FAREAST\.autorest\@autorestcore@3.1.3\nodemodules\@autorest\core\dist' (3.1.3)
INFORMATION: > Loading local AutoRest extension '@autorest/java' (C:\github\autorest.java)
INFORMATION: > Loading local AutoRest extension '@autorest/java.vanilla' (C:\github\autorest.java\javagen)
INFORMATION: > Loading local AutoRest extension '@autorest/java.preprocessor' (C:\github\autorest.java\javagen\..\preprocessor)
INFORMATION: > Loading local AutoRest extension '@autorest/java.postprocessor' (C:\github\autorest.java\javagen\..\postprocessor)
INFORMATION: > Loading AutoRest extension '@autorest/modelerfour' (4.21.4->4.21.4)
WARNING (PreCheck/CheckDuplicateSchemas): Checking for duplicate schemas, this could take a (long) while.  Run with --verbose for more detail.
INFORMATION (Postprocessor): README found at: file:///C:/github/autorest.java/readme.md
[10.84 s] Generation Complete
```

**Comment by weidongxu-microsoft** [comment]
maybe it because the node.js version. I remember it require something new e.g. 14.
